File: coding/12th December 2025/Destination City.java
//Destination City
class Solution 
{
    public String destCity(List<List<String>> paths) 
    {
        String res = paths.get(0).get(0);
        HashMap<String, String> hmap = new HashMap<String, String>();
        for(List<String> l : paths)
            hmap.put(l.get(0), l.get(1));
        while(hmap.containsKey(res))
            res = hmap.get(res);
        return res;
    }
}

File: Coding in Python/17th March 2026/Destination City.py
#Destination City
class Solution:
    def destCity(self, paths: List[List[str]]) -> str:
        result = paths[0][0]
        map = dict()
        for arr in paths:
            map[arr[0]] = arr[1]
        while result in map:
            result = map[result]
        return result
